File: app/src/main/java/com/rhm/pwn/model/URLCheckChangeNotifier.java
package com.rhm.pwn.model;

import java.util.Observable;

/**
 * Created by sambo on 8/29/2017.
 */

public class URLCheckChangeNotifier extends Observable {
    private static URLCheckChangeNotifier instance = new URLCheckChangeNotifier();

    public void update(Boolean b) {
        setChanged();
        notifyObservers(b);
    }

    public static URLCheckChangeNotifier getNotifier() {
        return instance;
    }
}

File: app/src/main/java/com/rhm/pwn/model/URLCheckChangeNotifier.kt
package com.rhm.pwn.model

import java.util.Observable

/**
 * Created by sambo on 8/29/2017.
 */

class URLCheckChangeNotifier : Observable() {

    fun update(b: Boolean?) {
        setChanged()
        notifyObservers(b)
    }

    companion object {
        private val notifier = URLCheckChangeNotifier()

        @JvmStatic
        fun getNotifier() : URLCheckChangeNotifier {
            return notifier
        }
    }
}
